llvm/lib/Target/SPIRV/SPIRVUtils.cpp

```diff
@@ -245,3 +246,95 @@ bool isSpvIntrinsic(MachineInstr &MI, Intrinsic::ID IntrinsicID) {
 Type *getMDOperandAsType(const MDNode *N, unsigned I) {
   return cast<ValueAsMetadata>(N->getOperand(I))->getType();
 }
+
+// The set of names is borrowed from the SPIR-V translator.
+// TODO: may be implemented in SPIRVBuiltins.td.
+static bool isPipeOrAddressSpaceCastBI(const StringRef MangledName) {
```

Comment:
Just a question - do we really need to have full list of the names? Wouldn't something like `MangledName.startswith("work_group") work if we already know it's a BI?

Reply:
Maybe it would be acceptable for compiling our tests. However when we call isOclOrSpirvBuiltin() interface, we don't know if it's really a BI name, so the interface helps to detect BI and demangle its name.

In other words if we just check MangledName.startswith("work_group"), "work_group_is_not_builtin" would be also qualified as a BI, but it's incorrect.